runtime/src/epoch_stakes_service.rs

```diff
@@ -0,0 +1,66 @@
+use {
+    crate::{
+        bank::Bank,
+        epoch_stakes::{BLSPubkeyToRankMap, EpochStakes},
+    },
+    crossbeam_channel::Receiver,
+    log::warn,
+    parking_lot::RwLock as PlRwLock,
+    solana_sdk::{
+        clock::{Epoch, Slot},
+        epoch_schedule::EpochSchedule,
+    },
+    std::{collections::HashMap, sync::Arc, thread},
+};
+
+struct State {
+    stakes: HashMap<Epoch, EpochStakes>,
+    epoch_schedule: EpochSchedule,
+}
+
+/// A service that regularly updates the epoch stakes state from the bank forks
+/// and exposes various methods to access the state.
+pub struct EpochStakesService {
+    state: Arc<PlRwLock<State>>,
+}
+
+impl EpochStakesService {
+    pub fn new(new_bank_receiver: Receiver<Arc<Bank>>) -> Self {
```

Comment:
lets take in the current root bank here to initialize with, just so we can serve data until the first epoch rolls over.

Reply:
good idea.  Now added.